### DISCUSSION TITLE
Connect to Gsheet using a service key (json file)

### DISCUSSION THREAD

**Discussion by agavrel** [Byron/google-apis-rs#437] [Q&A] [answered]
I didn't see any documentation related to this kind of authentication, did I miss it or is it a missing feature?

**Comment by Byron** [comment, marked as answer]
[`yup-oauth2`](https://crates.io/crates/yup-oauth2) is handling authentication and the [installed flow](https://docs.rs/yup-oauth2/8.3.0/yup_oauth2/#installed-flow-usage) is probably what you are looking for.

**Comment by agavrel** [comment]
It doesn't seem to be the installed flow as I am talking about a service key: "Custom { kind: InvalidData, error: "Unknown application secret format" }'"

Ok the answer is in the crate introduction actually:

"An oauth2 implementation, providing the 'device', 'service account' and 'installed' authorization flows"
